AnnoDesigner/AnnoCanvas.xaml.cs

```diff
@@ -761,163 +785,212 @@ private void RenderObjectInfluenceRadius(DrawingContext drawingContext, List<Lay
+        {
+            var stroked = true;
+            var smoothJoin = true;
 
-                        //Following the rules for quadrant 2 - go right and down
-                        var currentPoint = new Point(startPoint.X, startPoint.Y);
-                        while (endPoint != currentPoint)
-                        {
-                            currentPoint = new Point(currentPoint.X, currentPoint.Y + 1);
-                            sgc.LineTo(_coordinateHelper.GridToScreen(currentPoint, GridSize), stroked, smoothJoin);
-                            currentPoint = new Point(currentPoint.X + 1, currentPoint.Y);
-                            sgc.LineTo(_coordinateHelper.GridToScreen(currentPoint, GridSize), stroked, smoothJoin);
-                        }
+            var geometryFill = true;
+            var geometryStroke = true;
 
-                        ////////////////////////////////////////////////////////////////
-                        startPoint = endPoint;
-                        //Draw in height of object
-                        sgc.LineTo(_coordinateHelper.GridToScreen(new Point(startPoint.X, bottomRightCorner.Y), GridSize), stroked, smoothJoin);
+            var cellsInInfluenceRange = RoadSearchHelper.BreadthFirstSearch(
+                PlacedObjects.Select(o => o.WrappedAnnoObject),
+                Enumerable.Repeat(curLayoutObject.WrappedAnnoObject, 1),
+                o => o.InfluenceRange - 1,// reduce distance to get cells INSIDE influence range (not those which are touching the influence range)
+                gridDictionary: gridDictionary);
 
-                        //Draw quadrant 3
-                        //Get end value to draw from top-left of 3rd quadrant to bottom-left of 3rd quadrant
-                        //Move startPoint to bottomLeftCorner (x value is already correct)
-                        startPoint = new Point(startPoint.X, bottomRightCorner.Y);
-                        endPoint = new Point(bottomRightCorner.X, bottomRightCorner.Y + influenceRange);
+            var points = PolygonBoundaryFinderHelper.GetBoundaryPoints(cellsInInfluenceRange);
 
-                        //Following the rules for quadrant 3 - go left and down
-                        currentPoint = new Point(startPoint.X, startPoint.Y);
-                        while (endPoint != currentPoint)
-                        {
-                            currentPoint = new Point(currentPoint.X - 1, currentPoint.Y);
-                            sgc.LineTo(_coordinateHelper.GridToScreen(currentPoint, GridSize), stroked, smoothJoin);
-                            currentPoint = new Point(currentPoint.X, currentPoint.Y + 1);
-                            sgc.LineTo(_coordinateHelper.GridToScreen(currentPoint, GridSize), stroked, smoothJoin);
-                        }
+            sgc.BeginFigure(_coordinateHelper.GridToScreen(new Point(points[0].X, points[0].Y), GridSize), geometryFill, geometryStroke);
```

Comment:
Why not use the already existing `Point`?
Like: `_coordinateHelper.GridToScreen(points[0], GridSize)`

Reply:
Yes, I'll definitely. I just didn't notice it when I refactored it from `(int, int)`.